vcr/matchers.py

```diff
@@ -30,12 +32,29 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
+        r2_body  = r2.body.read()
+    else:
+        r1_body = r1.body
+        r2_body  = r2.body
+    if r1.headers.get('Content-Type') == r2.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
```

Comment:
I'm not necessarily against this idea, but the point is that this PR can not go in to master as is, because its behavior is inconsistent with respect to well formed content-type headers.

You'll have to convince @kevin1024 that this is the right thing to do since he said:
> I can picture someone wanting to write a custom matcher where they care about header capitalization, so I don't think I want to change the matcher API - just use the CaseInsensitiveDict for looking up the content-type of our "smart body matcher".

I can see the merits of both approaches, so I'm not going to take a side on this point, but I think that the capitlization issue definitely needs to be addressed in the content-type body matcher in some way or another.